ql/instruments/bonds/amortizingcmsratebond.hpp

```diff
@@ -48,8 +48,24 @@ namespace QuantLib {
                     const std::vector<Rate>& floors = {},
                     bool inArrears = false,
                     const Date& issueDate = Date());
-    };
 
+        AmortizingCmsRateBond(
+                    Natural settlementDays,
+                    const std::vector<Real>& notionals,
+                    const std::vector<Real>& redemptions,
```

Comment:
The vector of redemptions could be added as a last optional parameter to the existing constructor.  This would reduce duplication and would also make it easier to support keyword arguments in Python and other languages.

The same goes for the other classes.